packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -38,33 +43,98 @@ type DropdownCoreProps = {
 };
 
 export default class DropdownCore extends React.Component<DropdownCoreProps> {
+    node: ?HTMLDivElement;
+    // TODO(sophie): figure out flow typing
+    handleClick: (event: any) => void;
+
     static defaultProps = {
         alignment: "left",
     };
 
+    constructor(props: DropdownCoreProps) {
+        super(props);
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    componentWillMount() {
+        document.addEventListener("click", this.handleClick, false);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener("click", this.handleClick, false);
+    }
+
+    handleClick(event: SyntheticEvent<>) {
+        // If click was inside the component, return
+        // $FlowFixMe
+        if (this.node && this.node.contains(event.target)) {
+            return;
+        }
+        this.props.onClose();
+    }
+
     render() {
         const {alignment, items, open, opener, style} = this.props;
 
+        // TODO(sophie): why doesn't ref work with View?
```

Comment:
Because it's not a DOM element.  You'll need to call `ReactDOM.findDOMNode` on the `ref` to get the underlying DOM node.  We may change this in the future after we upgrade to React 16.

Reply:
Please use `ref` on the `View`.